### PR TITLE
allow phantomjs to start up with ssl errors ignored

### PR DESCRIPTION
**Pull request by jberney**

### CODE REVIEW

File: src/lib/headless.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 const lazypipe = require('lazypipe');
-const {listen} = require('./server');
+const server = require('./server');
```
Comment (jberney) [inline src/lib/headless.js]: Have to keep the `listen` method bound to the `server` object for spying

File: src/lib/headless.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 const once = require('lodash.once');
 const path = require('path');
 const portfinder = require('portfinder');
 const qs = require('qs');
-const {spawn} = require('child_process');
+const childProcess = require('child_process');
```
Comment (jberney) [inline src/lib/headless.js]: Have to keep the `spawn` method bound to the `childProcess` object for spying

File: src/lib/headless.js [hunk 3]
```diff
@@ -36,15 +36,15 @@ function onError(message) {
...
+  if (findOpenPort) return getPort().then(port => server.listen(port, files, options));
+  return server.listen(port, files, options);
 }
 
 function defaultReporters(options, profile) {
   return [new TerminalReporter(options), profile && new ProfileReporter(options)].filter(Boolean);
 }
 
-function createServer(options) {
+function createServer(options = {}) {
```
Comment (jberney) [inline src/lib/headless.js]: Defaulting options to avoid a NPE check later

File: src/lib/headless.js [hunk 4]
```diff
@@ -53,7 +53,9 @@ function createServer(options) {
     .pipe(() => through((files, enc, next) => getServer(files, options).then(i => next(null, i))))
     .pipe(() => flatMap(({server, port}, next) => {
       const stdio = ['pipe', output === 'stdout' ? 'pipe' : 1, output === 'stderr' ? 'pipe' : 2];
-      const phantomProcess = spawn(command, [runner, port, query], {cwd: path.resolve(__dirname), stdio});
+      const args = [runner, port, query];
+      options.ignoreSslErrors && args.splice(0, 0, '--ignore-ssl-errors=true');
```
Comment (jberney) [inline src/lib/headless.js]: If the user wants to ignore ssl errors, prepend the phantom flag to the args

**Comment by rdy** [comment]
If you want I can add this PR but I don't know if you care about phantomjs support and SSL at this point.

**Pull request closed without merging**